### ISSUE TITLE
Ej vectADT

### ISSUE THREAD

**Issue by ssommerr** [PI-ITBA/2024_02#177]
Hola, como estan? Tenia una consulta con el ejercicio 11 de la guia que me reaparecio en un parcial. En la solucion plantea hacer un vector de estructuras que dijera si esta ocupado o no el espacio, pero yo habia pensado en hacer en vez de una estructura un vector dinamico de elemtype y en el caso de que se encontrara vacia la posicion ponerle 0 (teniendo un size y un dim) Por que estaria mal?

**Comment by ssommerr** [comment]
A la vez plantee un elemtype *vector y no estoy segura si deberia ser un elemtype **vector, me estoy empezando a confundir :/

**Comment by ssommerr** [comment]
Hago una consulta más, en este ejercicio: Social ADT
#ifndef PI_SOCIALADT_H
#define PI_SOCIALADT_H

typedef struct socialCDT * socialADT;

/* Crea un nuevo TAD vacío */
socialADT newSocial();

/* Libera todos los recursos reservados por el TAD */
void freeSocial(socialADT soc);

/* Almacena una nueva persona. Si la persona existe, no hace nada
** Guarda una copia del nombre, no simplemente el puntero
*/
void addPerson(socialADT soc, const char * name);

/* Si existe una persona con ese nombre, agrega la nueva relación
** Si la persona no existe, no hace nada
** Si related ya estaba relacionado, lo agrega repetido
** Almacena una copia de related, no simplemente el puntero
**
*/
void addRelated(socialADT soc, const char * name, const char * related);

/* Retorna una copia de los nombres relacionados con una persona
** en orden alfabético.
** Para marcar el final, después del último nombre se coloca NULL
** Si la persona no existe, retorna un vector que sólo tiene a NULL como
** elemento
*/
char ** related(const socialADT soc, const char * person);

/* Retorna una copia de los nombres de las personas en orden alfabético.
** Para marcar el final, después del último nombre se coloca NULL
** Si no hay personas, retorna un vector que sólo tiene a NULL como
** elemento
*/
char ** persons(const socialADT soc);


#endif //PI_SOCIALADT_H


mi idea era crear el siguiente CDT
typedef struct alfabetoCDT{
struct nombres vector[cantletras]
}
typedef struct nombres{
char *nombre
struct related * firstrelated
struct nombres *tail
}
typedef struct  related{
char *nombre
struct related *tail
size-t repeticiones
}

estaria mal? senti que si ordeno de esta forma puedo buscar mas rapidamente si aparecio la persona para agregarla, a su vez no necesita ser un puntero a estructura porque si lo inciializo en calloc y chequeo que char* nombre==NULL sigifica que no hay personas con esa letra. Lo que capaz esta de mas es el de repeticiones ya que no te pide nada con eso, pero me parecio agregarlo ya que dice que se mete 2 veces a la persona si ya aparecio por lo que para otra funcion serviria

**Comment by marcelogarberoglio** [comment]
El código que agregaste es del socialADT, no de vectorADT.
Si hacés un vector de elemType no le podés asignar cero, o sea
elenType * vec;
--------
vec[i] = 0;    NO se puede hacer, sirve únicamente si elemType es int o char, long, etc. Si elemType es un struct, no va a compilar.

**Comment by ssommerr** [comment]
Mi segunda preg iba por el ej de social, y del anterior estaria muy mal hacer un elemtype **vec? Perdon se que el parcial es mañana:(

**Comment by marcelogarberoglio** [comment]
no, no estaría mal hacer vector de punteros a elemType. Es una de las alternativas que discutimos en clase. En ese caso en cada posición que está libre podés tener NULL.
